addon-test-support/-private/filter-test-modules.js

```diff
@@ -0,0 +1,51 @@
+// A regular expression to help parsing a string to verify regex.
+const MODULE_PATH_REGEXP = /^(!?)\/([\w\W]*)\/(i?$)/;
```

Comment:
I believe the following RegExp, maybe a better fit: `/^(!?)\/(.*)\/(i?)$/`

* `$` is at the end
* the character class used inside the `/` and `/` is less restrictive, supporting literally anything. Which supports users providing there own character classes, and other such things.